community_server/src/Controller/AppRequestsController.php

```diff
@@ -265,6 +266,89 @@ public function createCoins()
         
     }
     
+    public function getBalance($session_id = 0)
+    {
+
+        $login_result = $this->requestLogin($session_id, false);
+        if($login_result !== true) {
+            return $this->returnJson($login_result);
+        }
+        $session = $this->getRequest()->getSession();
+        $user = $session->read('StateUser');
+        $state_balances_table = TableRegistry::getTableLocator()->get('StateBalances');
+        $state_balances_table->updateBalances($user['id']);
+        
+        $state_balance = $state_balances_table->find()->where(['state_user_id' => $user['id']])->first();
+
+        if(!$state_balance) {
+            return $this->returnJson(['state' => 'success', 'balance' => 0]);
+        }
+        $now = new FrozenTime();
+        return $this->returnJson([
+            'state' => 'success',
+            'balance' => $state_balance->amount,
+            'decay' => $state_balance->partDecay($now),
+            'decay_date' => $now
+        ]);
+    }
+    
+    public function listTransactions($page = 1, $count = 25, $orderDirection = 'ASC', $session_id = 0)
+    {
+        $startTime = microtime(true);
+        $login_result = $this->requestLogin($session_id, false);
+        if($login_result !== true) {
+            return $this->returnJson($login_result);
+        }
+        $session = $this->getRequest()->getSession();
+        $user = $session->read('StateUser');
+        
+        
+        $stateBalancesTable = TableRegistry::getTableLocator()->get('StateBalances');
+        $stateUserTransactionsTable = TableRegistry::getTableLocator()->get('StateUserTransactions');
+        $transactionsTable  = TableRegistry::getTableLocator()->get('Transactions');
+        
+        
+        $stateBalancesTable->updateBalances($user['id']);
+
+        $gdtSum = 0;        
+        $gdtEntries = $this->JsonRequestClient->sendRequestGDT(['email' => $user['email']], 'GdtEntries' . DS . 'sumPerEmailApi');
+
+        if('success' == $gdtEntries['state'] && 'success' == $gdtEntries['data']['state']) {
```

Comment:
Why?
I use this style as often I remember to prevent spelling errors with the ==.
With 
```php
if('success' == $gdtEntries['state'])
```
you cannot forget a = and write:
```php 
if('success' = $gdtEntries['state']) 
```
because this will trigger an error in compiler/interpreter
while this: 
```php
 if($gdtEntries['state'] = 'success')
```
will lead to a hard to find bug.

Reply:
ok, that makes alot of sense, I did not know that - ty ❤️